scripts/mongodbintegrationmvp.py

```diff
@@ -0,0 +1,230 @@
+                model_kwargs={"seed": 999},  # For reproducibility
+            ),
+            GenerationStep(
+                model=Models.GPEI,
+                num_trials=-1,
+                max_parallelism=3,
+                model_kwargs={},
+            ),
+        ]
+    )
+    ax_client = AxClient(generation_strategy=generation_strategy)
+    ax_client.create_experiment(
+        name=obj1_name, parameters=parameters, objectives=objectives
+    )
+    print(f"Created new experiment with {SOBOL_TRIALS} Sobol trials")
+
+    # Save initial snapshot
+    save_ax_snapshot_to_mongodb(ax_client, obj1_name)
+else:
+    print("Resuming existing experiment")
+
+# Get current trial count to determine how many more trials to run
+current_trials = ax_client.get_trials_data_frame()
+start_trial = len(current_trials) if current_trials is not None else 0
+
+print(f"Starting optimization: running trials {start_trial} to {MAX_TRIALS-1}")
+
+for i in range(start_trial, MAX_TRIALS):
+    try:
+        # Get next trial
+        parameterization, trial_index = ax_client.get_next_trial()
+
+        # Extract parameters
+        x1 = parameterization["x1"]
+        x2 = parameterization["x2"]
+
+        print(f"Trial {trial_index}: x1={x1:.3f}, x2={x2:.3f}")
+
+        # Save snapshot before running experiment (preserves pending trial)
+        save_ax_snapshot_to_mongodb(ax_client, obj1_name)
+
+        # Evaluate objective function
+        results = branin(x1, x2)
+
+        # Format raw_data as expected by AxClient
+        raw_data = {obj1_name: results}
+
+        # Complete trial
+        ax_client.complete_trial(trial_index=trial_index, raw_data=raw_data)
+
+        # Save snapshot after completing trial
+        save_ax_snapshot_to_mongodb(ax_client, obj1_name)
+
+        # Get current best for progress tracking
+        try:
+            best_parameters, best_metrics = ax_client.get_best_parameters()
+            best_value = best_metrics[0][obj1_name]
+            print(
+                f"Trial {trial_index}: result={results:.3f} | "
+                f"Best so far: {best_value:.3f}"
+            )
+        except Exception:
+            print(f"Trial {trial_index}: result={results:.3f}")
+
+    except Exception as e:
+        print(f"Error in trial {trial_index}: {e}")
+        continue
+
+print("\nOptimization completed!")
+try:
+    best_parameters, best_metrics = ax_client.get_best_parameters()
+    print(f"Best parameters: {best_parameters}")
+    print(f"Best metrics: {best_metrics}")
+
+    # Save final snapshot
+    save_ax_snapshot_to_mongodb(ax_client, obj1_name)
+
+    # Print experiment summary
+    trials_df = ax_client.get_trials_data_frame()
+    if trials_df is not None:
+        print(f"Total trials completed: {len(trials_df)}")
+        print(f"Best objective value: {trials_df[obj1_name].min():.6f}")
+
+except Exception as e:
+    print(f"Error getting best parameters: {e}")
+
+# Clean up MongoDB connection
+mongo_client.close()
```

Comment:
Good to close the client. I suppose a top-level try-except-finally could be implemented at some point if we find that lots of mongo connections are being leftover during restarts, but I think probably not an issue. Again, just a musing